src/SFA.DAS.AssessorService.Application/Handlers/OrganisationStandards/OrganisationStandardVersionOptInHandler.cs

```diff
@@ -63,10 +63,20 @@ public async Task<OrganisationStandardVersion> Handle(OrganisationStandardVersio
                 };
 
                 var existingVersion = await _repository.GetOrganisationStandardVersionByOrganisationStandardIdAndVersion(orgStandard.Id, request.Version);
-                if (existingVersion != null)
-                    throw new InvalidOperationException("OrganisationStandardVersion already exists");
 
-                await _repository.CreateOrganisationStandardVersion(entity);
+                if (request.OptInFollowingWithdrawal)
```

Comment:
```
                if (request.OptInFollowingWithdrawal && existingVersion != null)
                {
                    await _repository.UpdateOrganisationStandardVersion(entity);
                }

                else if(existingVersion != null)
                {
                        throw new InvalidOperationException("OrganisationStandardVersion already exists");
                }

               else
               {
                      await _repository.CreateOrganisationStandardVersion(entity);
               }
```


Think the logic needs tweaking as this could mean some bits fall through the gaps?

- if existing version not null and following withdrawal, we can update
- if existing version not null but not following a withdrawal, we throw
- if existing version is null, we can create.


in the existing logic, you could set opt in following withdrawal to true, but existing version could be null and nothing happens. Removing nesting is better for readability

Reply:
Agreed - Tidied Up / Updated as requested